tools/telemetry-generator/test/memoryUsageTestHandler.test.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const handler = require("../src/handlers/memoryUsageTestHandler");
+
+describe("memoryUsageTestHandler", () => {
+	let mockLogger: any;
+	let mockFileData: any;
+
+	beforeEach(() => {
+		mockLogger = {
+			send: jest.fn(),
+		};
+
+		mockFileData = {
+			suiteName: "TestSuite",
+			benchmarks: [
+				{
+					benchmarkName: "Benchmark1",
+					customData: {
+						"Heap Used Avg": "123.45",
+						"Heap Used StdDev": "67.89",
+					},
+				},
+			],
+		};
+	});
+
+	it("should log endpoint name if defined", () => {
+		process.env.FLUID_ENDPOINTNAME = "testEndpoint";
+		const consoleLogSpy = jest.spyOn(console, "log").mockImplementation(() => {});
+
+		handler(mockFileData, mockLogger);
+
+		expect(consoleLogSpy).toHaveBeenCalledWith("ENDPOINTNAME", "testEndpoint");
+
+		consoleLogSpy.mockRestore();
+		delete process.env.FLUID_ENDPOINTNAME;
+	});
+
+	it("should log default endpoint name if not defined", () => {
+		const consoleLogSpy = jest.spyOn(console, "log").mockImplementation(() => {});
+
+		handler(mockFileData, mockLogger);
+
+		expect(consoleLogSpy).toHaveBeenCalledWith(
+			"ENDPOINTNAME not defined using local as default.",
+		);
+
+		consoleLogSpy.mockRestore();
+	});
```

Comment:
These two tests are "going too deep", I think. That logging is not a functional part of the handler so I'd say best to not have tests that _want_ us to log that value. If that message was something important about the logger behavior, I could see us having some tests for it, but in this case I'd recommend we just delete these tests, so no one gets confused in the future and thinks we need those log messages to always be there for some reason.

Reply:
removed